src/plugins/createReactEventPlugin.ts

```diff
@@ -0,0 +1,31 @@
+import { Plugin } from "prosemirror-state";
+import { EditorView } from "prosemirror-view";
+
+export interface DOMEventMap extends HTMLElementEventMap {
+  [event: string]: Event;
+}
+
+export type EventHandler<
+  EventType extends keyof DOMEventMap = keyof DOMEventMap
+> = (view: EditorView, event: DOMEventMap[EventType]) => boolean | void;
+
+export function createReactEventPlugin(
+  eventHandlerRegistry: Map<keyof DOMEventMap, Set<EventHandler>>
+) {
+  const domEventHandlers: Record<keyof DOMEventMap, EventHandler> = {};
+  for (const [event, handlers] of eventHandlerRegistry.entries()) {
+    function handleEvent(view: EditorView, event: Event) {
+      for (const handler of handlers) {
+        if (handler(view, event)) return true;
```

Comment:
Should we be using `useEditorEvent` under the hood to make the event handler reference stable while capturing other values in closures that are consistent with the current view state?